integrated_channels/integrated_channel/models.py

```diff
@@ -907,14 +920,21 @@ class IntegratedChannelAPIRequestLogs(TimeStampedModel):
     response_body = models.TextField(
         help_text="API call response body", blank=True, null=True
     )
-    channel_name = models.TextField(
+    channel_name = models.CharField(
         help_text="Name of the integrated channel associated with this API call log record.",
-        blank=True
+        max_length=32,
+        choices=CHANNEL_NAMES,
+        blank=True,
     )
 
     class Meta:
         app_label = "integrated_channel"
         verbose_name_plural = "Integrated channels API request logs"
+        indexes = [
+            models.Index(
+                fields=['channel_name', 'status_code', 'enterprise_customer_configuration_id', 'endpoint',],
+            )
+        ]
```

Comment:
Curious to know why did we choose these fields for indexing?

Reply:
@MueezKhan246 I think we can try to test the changes with the optimized query first. If that doesn't resolve the issue then we can look into adding these indexes. I'm just trying to be cautious just so we don't add any redundant indexes.

Reply:
Agree, let's test the changes with the optimized query first.

Reply:
@sameenfatima78 @zamanafzal i have removed the index for now

Reply:
Just approved your PR